widgetry/src/style/buttons.rs

```diff
@@ -0,0 +1,217 @@
+use super::ButtonStyle;
+use crate::{ButtonBuilder, ControlState, ScreenDims, Style};
+
+pub trait StyledButtons<'a> {
```

Comment:
I'm not sure how closely you looked at the Draft, but since then, I renamed the `Buttons` trait to `StyledButtons` and moved it from map_gui to widgetry.

Your preliminary feedback lead me to hang the button colors on widgetry::Style instead of map_gui::ColorScheme, which allowed me to avoid hardcoding lots of stuff in widgetry.